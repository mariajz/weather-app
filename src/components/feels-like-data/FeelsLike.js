import React from 'react';
import WeatherDetailsCard from '../../commons/weather-details-card';
import { getFeelsLikeText } from '../../utils/helpers';
import { degreeSymbol } from '../../constants/Constants';

const FeelsLike = ({ feelsLikeProps }) => {
    const { feelsLike } = feelsLikeProps;
    const { feelsLikeText } = getFeelsLikeText(feelsLike);
    const feelsLikeCardProps = {
        iconName: 'temperature',
        cardName: 'FEELS LIKE',
        title: `${feelsLike}${degreeSymbol}C`,
        subtitle: '',
        content: feelsLikeText,
    };

    return <WeatherDetailsCard weatherDetailsCardProps={feelsLikeCardProps} />;
};

export default FeelsLike;
