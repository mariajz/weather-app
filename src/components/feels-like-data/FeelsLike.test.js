import { render } from '@testing-library/react-native';
import React from 'react';
import FeelsLike from './FeelsLike';

const defaultProps = {
    feelsLikeProps: {
        feelsLike: 38,
    },
};

describe('FeelsLike', () => {
    it('should render FeelsLike', () => {
        const container = render(<FeelsLike {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
