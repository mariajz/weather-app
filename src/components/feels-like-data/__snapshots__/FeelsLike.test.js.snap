// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeelsLike should render FeelsLike 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255,255,255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "height": 170,
      "marginTop": 10,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": "49%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Image
      size={15}
      source={
        {
          "testUri": "../../../assets/icons/temperature.png",
        }
      }
      style={
        {
          "height": 15,
          "width": 15,
        }
      }
      testID="icon"
      tintColor="white"
    />
    <Text
      marginLeft={5}
      size={14}
      style={
        {
          "color": "white",
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      FEELS LIKE
    </Text>
  </View>
  <View
    style={
      {
        "height": 120,
        "justifyContent": "space-between",
        "marginTop": 8,
      }
    }
  >
    <View>
      <Text
        size={28}
        style={
          {
            "color": "white",
            "fontSize": 28,
            "marginLeft": 0,
          }
        }
      >
        38Â°C
      </Text>
      <Text
        size={20}
        style={
          {
            "color": "white",
            "fontSize": 20,
            "marginLeft": 0,
          }
        }
      />
    </View>
    <View />
    <Text
      size={10}
      style={
        {
          "color": "white",
          "fontSize": 10,
          "marginLeft": 0,
        }
      }
    >
      Air feels dry and crisp today as humidity levels remain low.
    </Text>
  </View>
</View>
`;
